### PR TITLE
ANN: Check if the asm macro is contained in an unsafe block or function

### LINKED ISSUE #6044: "Surround with unsafe block" for asm! macro
## Environment

* **IntelliJ Rust plugin version:** 0.3.129.3308-202
* **Rust toolchain version:** 1.46.0-nightly (16957bd4d 2020-06-30) x86_64-pc-windows-gnu
* **IDE name and version:** CLion 2020.2.1 (CL-202.6948.80)
* **Operating system:** Windows 10 10.0

## Problem description
There is no "Surround with unsafe block" suggession when using the asm! macro.
image

### PR DESCRIPTION
**Pull request by Stzx**
- Closed #6044
- Part of #6045

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/annotator/RsUnsafeExpressionErrorAnnotatorTest.kt [hunk 1]
```diff
@@ -132,4 +134,33 @@ class RsUnsafeExpressionErrorAnnotatorTest : RsAnnotatorTestBase(RsUnsafeExpress
...
+    """)
+
+    fun `test need unsafe asm macro expr`() = checkErrors("""
+       fn main() {
+            <error descr="use of `asm!` is unsafe and requires unsafe function or block [E0133]">asm!("nop")</error>
+       }
+    """)
+
+    @ProjectDescriptor(WithDependencyRustProjectDescriptor::class)
+    fun `test a repeatedly defined asm macro`() = checkErrors("""
```
Comment (Stzx) [inline src/test/kotlin/org/rust/ide/annotator/RsUnsafeExpressionErrorAnnotatorTest.kt:151]: @Undin Can you see this test case? My local test failed and CI failed. Did I make a mistake?

**Comment by Kobzol** [comment]
Just a heads-up that if https://github.com/intellij-rust/intellij-rust/pull/5928 get merged the `asm!` macro will be parsed and represented slightly differently by the plugin, but it will not be a big change in regard to this PR.

**Comment by Kobzol** [comment]
Why have you closed this? 🤔

**Comment by Stzx** [comment]
> Why have you closed this? 🤔

**Comment by Stzx** [comment]
Oh!

It's been a bit too long. I even forgot about it. So I accidentally deleted it when I was cleaning it up.  😅

**Comment by vlad20012** [comment]
@Stzx thanks!
I've pushed a few fixes here.
bors r+

**Comment by Stzx** [comment]
Oh, okay. This looks great, thanks for all your work.

**Pull request merged**